compiler/rustc_middle/src/dep_graph/dep_node.rs

```diff
@@ -248,6 +230,19 @@ pub mod dep_kind {
                 is_anon: contains_anon_attr!($($attrs)*),
                 is_eval_always: contains_eval_always_attr!($($attrs)*),
 
+                can_reconstruct_query_key: || {
+                    if contains_anon_attr!($($attrs)*) {
+                        return false;
+                    }
+
+                    $({
+                        erase!($tuple_arg_ty);
+                        return <query_keys::$variant<'_> as DepNodeParams<TyCtxt<'_>>>
+                            ::can_reconstruct_query_key()
+                    })*
```

Comment:
This seems suspicious. The tuple_arg_ty doesn't seem like it needs to be recursed here, $variant is the only thing that matters, right? But previously it looks like the code this replaced did expand to the tuple arg's ability to be reconstructed... which is odd, because I would expect the first one to return given this expansion, while my expectation would be that we need all of the arguments to be reconstructible, not just the first one.

(The next commit in this PR seems to replace this code with the suggested lack of iteration; maybe that's correct but it seems like different behavior?)

This is the old version, before this PR, for reference:

```rust
                        DepKind :: $variant => {
                            // tuple args
                            $({
                                return <$tuple_arg_ty as DepNodeParams<TyCtxt<'_>>>
                                    ::can_reconstruct_query_key();
                            })*

                            true
                        }
```